libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java

```diff
@@ -0,0 +1,188 @@
+        mockTelemetryChannel = Mockito.mock(TelemetryChannel.class);
+
+        TelemetryConfiguration telemetryConfiguration = new TelemetryConfiguration();
+        telemetryConfiguration.setInstrumentationKey("UNITTEST-INSTRUMENTATION-KEY");
+        telemetryConfiguration.setChannel(mockTelemetryChannel);
+        TelemetryClient telemetryClient = new TelemetryClient(telemetryConfiguration);
+
+        botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void nullTelemetryClientThrows() {
+        Assert.assertThrows(IllegalArgumentException.class, () -> { 
+            new BotTelemetryClientImpl(null); 
+        });
+    }
+
+    @Test
+    public void nonNullTelemetryClientSucceeds() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+
+        BotTelemetryClient botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void overrideTest() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+        MyBotTelemetryClient botTelemetryClient = new MyBotTelemetryClient(telemetryClient);
+    }
+
+    @Test
+    public void trackAvailabilityTest() {
+        Map<String, String> properties = new HashMap<>();
+        Map<String, Double> metrics = new HashMap<>();
+        properties.put("hello", "value");
+        metrics.put("metric", 0.6);
+
+        botTelemetryClient.trackAvailability(
+            "test",
+            OffsetDateTime.now(),
+            Duration.ofSeconds(1000), // TODO: use computer ticks
+            "run location",
+            true,
+            "message",
+            properties,
+            metrics);
+
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry availabilityTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", availabilityTelemetry.getName());
+            Assert.assertEquals("message", availabilityTelemetry.getData().getMessage());
+            Assert.assertEquals("value", availabilityTelemetry.getProperties().get("hello"));
+            Assert.assertEquals((Double) 0.6, availabilityTelemetry.getMetrics().get("metric"));
+
+            return null;
+        }).when(mockTelemetryChannel).send(Mockito.any(AvailabilityTelemetry.class));
+    }
+
+    @Test
+    public void trackEventTest() {
+        Map<String, String> properties = new HashMap<>();
+        properties.put("hello", "value");
+        Map<String, Double> metrics = new HashMap<>();
+        metrics.put("metric", 0.6);
+        
+        botTelemetryClient.trackEvent("test", properties, metrics);
+
+        Mockito.doAnswer(invocation -> {
+            EventTelemetry eventTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventTelemetry.getName());
+            Assert.assertEquals("value", eventTelemetry.getProperties().get("hello"));
+            Assert.assertEquals((Double) 0.6, eventTelemetry.getMetrics().get("metric"));
```

Comment:
Same

Reply:
the cast was necessary because removing the quotes and making the value a numeric one (0.6) arises an ambiguity error about two different types of Asserts.assertEquals, I think one of them of type (Object, Object). Without that cast to make it explicit I get an error